pkg/distributor/distributor.go

```diff
@@ -603,6 +663,8 @@ func (d *Distributor) Push(ctx context.Context, req *cortexpb.WriteRequest) (*co
 		return nil, httpgrpc.Errorf(http.StatusTooManyRequests, "ingestion rate limit (%v) exceeded while adding %d samples and %d metadata", d.ingestionRateLimiter.Limit(now, userID), validatedSamples, len(validatedMetadata))
 	}
 
+	d.ingestionRate.Add(int64(validatedSamples))
```

Comment:
Should we use `totalN` instead of `validatedSamples` like we do for the per-tenant limiter?

Reply:
I was considering that. Problem is that in ingester, we don't include metadata in this computation. I think we should be consistent – either include metadata in both, or not.

Reply:
Good point. I agree we need to be consistent. Not including it, means there's no limit applied to metadata, so I could intentionally skip this limit with remote write requests including only metadata. Including metadata both in distributor and ingester may be safer.